src/metabase/models/params/shared.cljc

```diff
@@ -82,15 +82,87 @@
     #"^thisyear$"                           (lib/describe-temporal-interval 0 :year)
     #"^(past|next)([0-9]+)([a-z]+)s~?$" :>> (fn [matches] (apply format-relative-date matches))))
 
+#?(:clj (def ^:private day-in (DateTimeFormatter/ofPattern "EEE"
+                                                           ;; always read in en locale
+                                                           (i18n.impl/locale "en"))))
+
+#?(:clj (def ^:private day-out (DateTimeFormatter/ofPattern "EEEE")))
+
+(defn- format-day [value locale]
+  #?(:cljs (let [t (.locale moment)
+                 _ (.locale moment "en") ;; always read in en locale
+                 s (-> value
+                       (moment "ddd")
+                       (.locale locale)
+                       (.format "dddd"))]
+             (.locale moment t)
+             s)
```

Comment:
It took me a second to figure this out. It really needs a better factoring, since this is pretty janky and it's not at all obvious what's going on at first.

There's `u.time/string->timestamp` as well, but I don't know if it's quite as flexible as this. We could add a `:locale` to its `options` map likewise.